gems/sorbet-runtime/lib/types/enum.rb

```diff
@@ -158,6 +158,11 @@ def to_json(*args)
     serialize.to_json(*args)
   end
 
+  sig {params(args: T.untyped).returns(T.untyped)}
+  def as_json(*args)
+    serialize.as_json(*args)
```

Comment:
Maybe:
```suggestion
    serialized_val = serialize
    return serialized_val unless serialized_val.respond_to?(:as_json)
    
    serialized_val.as_json(*args)
```
to work around the fact that `String` does not always have `as_json` defined on it?

Reply:
I'm weary of subtle differences in behaviour depending on whether ActiveSupport is included or not.

`String` won't have `as_json` only when ActiveSupport isn't included. In which case, `as_json` would never really be called anyway.

Perhaps we should only define this method conditionally when ActiveSupport is included? 

Or define the method, but raise an explicit error (`NotImplementedError`?) if it's not included?